homeworks/dmytro.tolbatov_dmytrotolbatov/homework_5/innerCopy.js

```diff
@@ -0,0 +1,18 @@
+const testObj = { b: 'c', d: { e: 'f' } };
+
+function innerCopy(obj) {
+    return Object.assign({}, obj);
+}
+
+function innerCopy2(obj) {
+    const objCopy = {};
+
+    Object.keys(obj).forEach((key) => {
+        objCopy[key] = obj[key];
+    });
```

Comment:
both of your functions will not perform deep copying, keep in mind that non-primitive value are assigned "by link", not "by value"

hint: this a good case for recursion